--- photon/PhotonPoseEstimator.cpp
+++ photon/PhotonPoseEstimator.cpp
@@ -76,21 +76,6 @@ PhotonPoseEstimator::PhotonPoseEstimator(frc::AprilTagFieldLayout tags,
   InstanceCount++;
 }

-PhotonPoseEstimator::PhotonPoseEstimator(frc::AprilTagFieldLayout tags,
-                                         PoseStrategy strat, PhotonCamera&& cam,
-                                         frc::Transform3d robotToCamera)
-    : aprilTags(tags),
-      strategy(strat),
-      camera(std::make_shared<PhotonCamera>(std::move(cam))),
-      m_robotToCamera(robotToCamera),
-      lastPose(frc::Pose3d()),
-      referencePose(frc::Pose3d()),
-      poseCacheTimestamp(-1_s) {
-  HAL_Report(HALUsageReporting::kResourceType_PhotonPoseEstimator,
-             InstanceCount);
-  InstanceCount++;
-}
-
 void PhotonPoseEstimator::SetMultiTagFallbackStrategy(PoseStrategy strategy) {
   if (strategy == MULTI_TAG_PNP_ON_COPROCESSOR ||
       strategy == MULTI_TAG_PNP_ON_RIO) {
