From 7b7e69a6b43bdbc21db7b72491cc0c2544864027 Mon Sep 17 00:00:00 2001
From: David Vo <auscompgeek@users.noreply.github.com>
Date: Sun, 16 Apr 2023 13:50:48 +1000
Subject: [PATCH] PhotonPoseEstimator: Add constructor taking unique_ptr

---
 photonlib/PhotonPoseEstimator.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/photonlib/PhotonPoseEstimator.cpp b/photonlib/PhotonPoseEstimator.cpp
index 5b7ff54853d2..f328e9aad874 100644
--- a/photonlib/PhotonPoseEstimator.cpp
+++ b/photonlib/PhotonPoseEstimator.cpp
@@ -67,10 +67,16 @@ PhotonPoseEstimator::PhotonPoseEstimator(frc::AprilTagFieldLayout tags,
       m_robotToCamera(robotToCamera),
       lastPose(frc::Pose3d()),
       referencePose(frc::Pose3d()),
       poseCacheTimestamp(-1_s) {}
 
+PhotonPoseEstimator::PhotonPoseEstimator(frc::AprilTagFieldLayout tags,
+                                         PoseStrategy strat,
+                                         std::unique_ptr<PhotonCamera> cam,
+                                         frc::Transform3d robotToCamera)
+    : PhotonPoseEstimator(tags, strat, std::move(*cam), robotToCamera) {}
+
 void PhotonPoseEstimator::SetMultiTagFallbackStrategy(PoseStrategy strategy) {
   if (strategy == MULTI_TAG_PNP) {
     FRC_ReportError(
         frc::warn::Warning,
         "Fallback cannot be set to MULTI_TAG_PNP! Setting to lowest ambiguity",
-- 
2.39.1

